TheWarWithin/ShamanElemental.lua

```diff
@@ -1092,10 +1088,27 @@ spec:RegisterCombatLogEvent( function( _, subtype, _,  sourceGUID, sourceName, _
 
         end
 
-        if ( subtype == "SPELL_DAMAGE" or subtype == "SPELL_PERIODIC_DAMAGE" ) and state.talent.elemental_equilibrium.enabled then
-            if bit.band( school, 4  ) == 1 then fireDamage   = GetTime() end
-            if bit.band( school, 16 ) == 1 then frostDamage  = GetTime() end
-            if bit.band( school, 8  ) == 1 then natureDamage = GetTime() end
+        if state.talent.elemental_equilibrium.enabled then
+            if ( subtype == "SPELL_DAMAGE" or subtype == "SPELL_PERIODIC_DAMAGE" ) then
+
+                if school == 4 then -- fire
+                    fireDamage = GetTime()
+                elseif school == 16 then -- frost
+                    frostDamage = GetTime()
+                elseif school == 8 then -- nature
+                    natureDamage = GetTime()
+                elseif school == 12 then -- volcanic
+                    fireDamage = GetTime()
+                    natureDamage = GetTime()
+                elseif school == 28 then -- elemental
+                    fireDamage = GetTime()
+                    frostDamage = GetTime()
+                    natureDamage = GetTime()
+                end
+            end
+            if subtype == "SPELL_AURA_APPLIED" and ( spellID == 378275 or spellID == 347348 ) then
```

Comment:
This can actually be else-if since the previous `if` and the current `if` are mutually exclusive.